recipe/meta.yaml

```diff
@@ -37,28 +37,52 @@ requirements:
     - libprotobuf {{ libprotobuf }}
     - libabseil {{ libabseil }}
     - pytest-runner 6.0.0
-    - numpy   1.21  # [py<310]
-    - numpy   1.23  # [py==310 or py==311]
-    - numpy   1.26  # [py>=312]
+    - numpy 2.0 # [py<313]
+    - numpy 2.1 # [py==313]
   run:
     - python
     - protobuf >=3.20.2
-    - {{ pin_compatible('numpy') }}
+    - numpy >=1.20,<2.3
     - typing-extensions >=3.6.2.1
     # through run_exports
     - libprotobuf
     - libabseil
 
+{% set tests_to_skip = "" %}
+# Skip test_regex_full_match and test_regex_invalid_pattern because google-re2 isn't available
+{% set tests_to_skip = tests_to_skip + "test_regex_full_match or test_regex_invalid_pattern" %}
+# Skip test_image_decoder_decode_jpeg_bgr_cpu, test_image_decoder_decode_jpeg_rgb_cpu, and test_image_decoder_decode_jpeg_grayscale_cpu
+# because of "AssertionError: Not equal to tolerance rtol=0.001, atol=1e-07",
+# see https://github.com/onnx/onnx/issues/4798
+{% set tests_to_skip = tests_to_skip + " or test_image_decoder_decode_jpeg_bgr_cpu or test_image_decoder_decode_jpeg_rgb_cpu or test_image_decoder_decode_jpeg_grayscale_cpu" %}
+
+# helper tests (test_to_array_with_external_data, test_make_tensor_raw) fail on big endian,
+# see https://github.com/onnx/onnx/issues/2992
+{% set tests_to_ignore = "" %}
+{% set tests_to_ignore = tests_to_ignore + " --ignore=onnx/test/helper_test.py" %}         # [linux and s390x]
+{% set tests_to_ignore = tests_to_ignore + " --ignore=onnx/test/test_external_data.py" %}  # [linux and s390x]
+
 test:
-  requires:
-    - pip
   imports:
     - onnx
+  source_files:
+    - onnx/test
```

Comment:
Tests are already part of the package AFAIK. You can `pytest --pyargs onnx` I think.
```suggestion
```

Reply:
The main obstacle is how to ignore tests, e.g., `--ignore=onnx/test/test_external_data.py` on `linux-s390x`.
Should I set `--ignore=test_external_data.py` or `--ignore=test/test_external_data.py`? Because the `test` folder is inside the `onnx` folder.

Reply:
I'll try to skip them instead of ignore

Reply:
You can `--ignore=$SP/onnx/test/test_external_data.py` for example.

Reply:
Fixed